src/circuit/operator/differentiatior.rs

```diff
@@ -0,0 +1,57 @@
+// Differentiator operator: direct implementation, without an explicit z
+
+use crate::algebra::GroupValue;
+use crate::circuit::operator_traits::{Operator, UnaryOperator};
+use std::borrow::Cow;
+
+pub struct Differentiator<V>
```

Comment:
I've been naming operators as actions ("Integrate", "Inspect", "Apply", etc.).  For consistency this would have to be `Differentiate`.  (I know there is also an "Integrator", but that one will go away before we merge this PR).